src/tools/clippy/tests/ui/borrow_interior_mutable_const/others.rs

```diff
@@ -10,7 +10,7 @@ use std::sync::Once;
 
 const ATOMIC: AtomicUsize = AtomicUsize::new(5);
 const CELL: Cell<usize> = Cell::new(6);
-const ATOMIC_TUPLE: ([AtomicUsize; 1], Vec<AtomicUsize>, u8) = ([ATOMIC], Vec::new(), 7);
+const ATOMIC_TUPLE: ([AtomicUsize; 1], Option<Box<AtomicUsize>>, u8) = ([ATOMIC], None, 7);
```

Comment:
What's the motivation of this Clippy test change? Is the empty `Vec` version now triggering the lint?

Reply:
The motivation is explained in the PR description and this Zulip topic that I started https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Hit.20an.20ICE.20in.20borrow_interior_mutable_const/near/446689692

Reply:
Oh right. I tried to fix that recently. But that's actually quite hard. Thanks for the pointer.